layer.yaml

```diff
@@ -1,31 +1,59 @@
-includes: ['layer:hadoop-client', 'interface:spark', 'interface:benchmark']
+includes:
+- layer:hadoop-client
+- interface:spark
+- interface:benchmark
 options:
-  hadoop-client:
+  apache-spark:
```

Comment:
I didn't see your reply while I was drafting the previous, but I think it addresses some of your questions.  However, to address explicitly why it would belong in `hadoop-client`, my thought in structuring it that way was that it was common functionality that could be provided by the base layer (i.e., so that not every client layer had to remember to call `dist_config.add_users()`, etc.), and layer options seemed the most natural way to send that data from the charm layer to the base layer.

Layer options was created as a way to provide a well defined, build-time validated way to pass data from charm layers to base layers, and to do away with the arbitrary yaml files like `dist.yaml` and `apache.yaml`.